### PR TITLE
feat: PickKeys

### PR DESCRIPTION
**Pull request by satanTime**
time to time I need to accept keys as parameters of a function which match only specific types in original object.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -571,6 +572,22 @@ type Result = RequiredKeys<T>;
...
+```typescript
+type T = {
+  a: number;
+  b?: string;
+  c: string | undefined;
+  d: string;
+};
+type Result = PickKeys<T, string>;
+// Result:
+// "d"
```
Comment (quezak) [inline README.md:588]: let's add another example with a complex type to better illustrate the possibilities
```suggestion
type Result = PickKeys<T, string>;
// Result: "d"
type Result2 = PickKeys<T, string | undefined>;
// Result2: "b" | "c" | "d"
```

**Review by quezak — APPROVED** [review]
Looks useful, I'm for it. @krzkaczor?

File: README.md (hunk 1 above)
Comment (satanTime) [reply README.md]: done

**Comment by satanTime** [comment]
Hi guys, any update on this?

**Comment by krzkaczor** [comment]
Thanks for this! Released as `ts-essentials@7.0.2`

**Comment by satanTime** [comment]
Thanks!

**Pull request merged**